src/server/main_service.cc

```diff
@@ -695,6 +695,8 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   config_registry.RegisterMutable("masterauth");
   config_registry.RegisterMutable("tcp_keepalive");
   config_registry.RegisterMutable("replica_partial_sync");
+  config_registry.RegisterMutable("max_eviction_per_heartbeat");
```

Comment:
well, now that you set it as Mutable config, your code that caches these flags in the c'tor is not correct

Reply:
but I now always update those two member variables in the eviction code, is it still not correct?

Reply:
in that case they do not need to be member variables and do not need to be set in the c'tor

Reply:
I kept them as part of the member variables as I felt those two parameters still logically are part of a db_slice object's property. With those defined in the class, the abstraction looks more complete to me...

Reply:
Ok, but in general I usually apply the following reasoning for similar cases: I ask myself whether the variable is part of the object state or not. We usually want to reduce the scope of variables (even on stack) to make the code more readable and structured. If we increase the scope of variables to more than it needs to be, then it's harder to reason about the correctness of the code since you need to track all the potential places they are being accessed.

Reply:
i will just take them out of the class... but we have to keep in mind that later if we start adding different policies, they very likely will come back to the class... or will become the member of a policy class whose instance becomes the member variable again of the `DbSlice`....